password.py

```diff
@@ -1,49 +1,302 @@
+
+    account_save()
+
+    return 
+
+# Asks user to pick a question and saves the answer
+def security_questions():
+     sec_questions_dict = {
+         "1":"What's your pet's name",
+         "2":"What's your mother's middle name",
+         "3":"What's your best friend's nickname",
+         "4":"Which hospital were you born in"
+      }
+
+     print("-"*26)
+     for key,val in sec_questions_dict.items():
+         print(key,":",val)
+     print("-"*26)
+
+     security_choice = input("Please choose a security question (Type in corresponding number):\n")
+
+     value = sec_questions_dict[security_choice]
+     print(value)
+     question_answer = input("")
+
+     account_info_dict["Security Question"] = value
+     account_info_dict["Security Answer"] = question_answer
+
+     return value, question_answer
 
 # Asks to generate a randomized password
 def password_randomizer_query():
+    accepted_answers = 'yes','no'
+
+    while True:
+        rand_answer = input("Do you want a randomized password? ('yes' or 'no'):\n")
+        if rand_answer == 'yes':
+            gen_pass = random_pass()
+            account_info_dict["Password"] = gen_pass
+            print("Your random pass is: " + gen_pass)
+            break
+        elif rand_answer == 'no':
+            break
+        elif rand_answer != accepted_answers:
+            print("Incorrect use.")
+
+    return rand_answer
 
-    return
 # Generates a randomized password using letters and digits
 def random_pass():
+    string = shane_char.ascii_characters_and_num()
+    letters_and_num = string.letters + string.digits
+    rand_password = random.choice(string.lowercase)
+    rand_password += random.choice(string.uppercase)
+    rand_password += random.choice(string.digits)
+    rand_password += random.choice(string.special_characters)
+
+    for characters in range(6):
+        rand_password += random.choice(letters_and_num)
+
+    password_list = list(rand_password)
+    random.SystemRandom().shuffle(password_list)
+    rand_password = ''.join(password_list)
+
+    return rand_password
 
-    return
 # Saves and stores account
 def account_save():
 
-    return
+    account_file = open('account.txt','w')
+    account_file.write(str(account_dict))
+    account_file.close()
+
+    return account_dict
+
+#lists account dict
+# def list_account():
+#     account_file_read = open('account.txt','r')
+#     print(account_file_read.read())
+#     account_file_read.close()
+
 # Recover exisiting account
 def account_recovery():
+    account_dict_file = open('account.txt','r')
+
+    for line in account_dict_file:
+        word = re.split('\{|: |, |\}', line)
+        name = word[1]
+        username = name.replace("'","")
+        pass_w = word[6]
+        password = pass_w.replace("'","")
+        sec_ques = word[8]
+        security_question = sec_ques.replace('"',"")
+        sec_answ = word[10]
+        security_answer = sec_answ.replace("'","")
+    
+    attempts = 4
```

Comment:
Impressive that you pretty much attempted to write the python interpreter again, but unnecessary. 

Doing things this way makes things extremely brittle (think about what happens when you have to add any other field to this program? Then you'd have to print this whole thing out and re-calculate everything in this function). 

On top of that, if I'm not mistaken python does not guarantee dictionary order, so on any given run of your program your `word` list could be in a completely different order than you expect.

Better to save and load in json and be able to use python as normal to make changes. If you need help with implementing this we can chat.

Reply:
In terms of the dictionary order, I made sure it would always be in those specific positions by printing out the file as a string earlier in the program; so the file would always have it in those positions. 

But yeah you're right, its very brittle and doesn't have flexible use at all.

Reply:
Even if you stringify it its not guaranteed to be in a certain order: https://portingguide.readthedocs.io/en/latest/dicts.html#changed-key-order

So for reasons only python knows about they can change the order of dictionary keys at anytime. Hence relying on a certain order is not safe.

Reply:
That link literally explained everything in both cases for me, thanks.